dockstore-integration-testing/src/test/java/io/dockstore/webservice/TokenResourceIT.java

```diff
@@ -256,6 +258,158 @@ public void testNinjaedGitHubUser() throws Exception {
+
+        // Login with Google still works
+        io.swagger.client.model.Token token = unAuthenticatedTokensApi.addGoogleToken(satellizerJSON);
+        Assert.assertEquals(100l, (long)token.getUserId());
+        // Login with GitHub still works
+        io.swagger.client.model.Token fakeGitHubCode = unAuthenticatedTokensApi.addToken(satellizerJSON);
+        Assert.assertEquals(100l, (long)fakeGitHubCode.getUserId());
+    }
+
+    private void registerAndLinkUnavailableTokens(TokensApi unAuthenticatedTokensApi) throws Exception {
+        // Should not be able to register new Dockstore account when profiles already exist
+        registerNewUsersWithExisting(unAuthenticatedTokensApi);
+        // Can't link tokens to other Dockstore accounts
+        addUnavailableGitHubTokenToGoogleUser(unAuthenticatedTokensApi);
+        addUnavailableGoogleTokenToGitHubUser();
+    }
+
+    @Test
+    public void recreateAccountsAfterSelfDestruct() throws Exception {
+        TokensApi unAuthenticatedTokensApi = new TokensApi(getWebClient(false, "n/a"));
+        createAccount1(unAuthenticatedTokensApi);
+        registerNewUsersAfterSelfDestruct(unAuthenticatedTokensApi);
+    }
+
+    /**
+     * Creates the Account 1: Google-created Dockstore account that is called GOOGLE_ACCOUNT_USERNAME1 but then changes to CUSTOM_USERNAME2
+     * and has the GOOGLE_ACCOUNT_USERNAME1 Google account linked and CUSTOM_USERNAME1 GitHub account linked
+     * @param unAuthenticatedTokensApi
+     * @throws Exception
+     */
+    private void createAccount1(TokensApi unAuthenticatedTokensApi) throws Exception {
+        mockGoogleHelper(GOOGLE_ACCOUNT_USERNAME1);
+        io.swagger.client.model.Token account1DockstoreToken = unAuthenticatedTokensApi
+                .addGoogleToken(satellizerJSONForRegistration);
+        Assert.assertEquals(GOOGLE_ACCOUNT_USERNAME1, account1DockstoreToken.getUsername());
+        mockGitHub(CUSTOM_USERNAME1);
+        TokensApi mainUserTokensApi = new TokensApi(getWebClient(true, GOOGLE_ACCOUNT_USERNAME1));
+        mainUserTokensApi.addGithubToken("fakeGitHubCode");
+    }
+
+    private void createAccount2(TokensApi unAuthenticatedTokensApi) throws Exception {
+        mockGoogleHelper(GOOGLE_ACCOUNT_USERNAME2);
+        io.swagger.client.model.Token otherGoogleUserToken = unAuthenticatedTokensApi.addGoogleToken(satellizerJSONForRegistration);
+        Assert.assertEquals(GOOGLE_ACCOUNT_USERNAME2, otherGoogleUserToken.getUsername());
+    }
+
+    /**
+     *
+     * @throws Exception
+     */
+    private void registerNewUsersWithExisting(TokensApi unAuthenticatedTokensApi) throws Exception {
+        mockGitHub(CUSTOM_USERNAME1);
+        mockGoogleHelper(GOOGLE_ACCOUNT_USERNAME1);
+        // Cannot create new user with the same Google account
+        try {
+            unAuthenticatedTokensApi.addGoogleToken(satellizerJSONForRegistration);
+            Assert.fail();
+        } catch (Exception e){
```

Comment:
Use the narrowest exception required, I think in this case you just need ApiException

Reply:
It's the mockGitHub and powermock that's throwing the Exception.

Reply:
@garyluu If that's true, the test is probably not quite testing what we want. 
We want to be testing against exceptions that are thrown in our code (or the swagger generated classes when expected). We shouldn't be testing in what situations powermock crashes with an exception. Feel free to swing by if you need clarification

Reply:
nvm, was looking at wrong line